### PR TITLE
chore(python): detect when samples testing requires custom dockerfile

### PR DESCRIPTION
**Pull request by parthea**
The [python-recaptcha-enterprise](https://github.com/googleapis/python-recaptcha-enterprise) requires a custom Dockerfile for samples testing. This PR eliminates the need to set the `custom_samples_dockerfile` argument of `py_library` which is currently done in owlbot.py [here](https://github.com/googleapis/python-recaptcha-enterprise/blob/main/owlbot.py#L42). IOW, kwargs `custom_samples_dockerfile` will automatically be set to `True` if the following file exists: `/.kokoro/docker/samples/Dockerfile`.

I tested the changes using the following commands after cloning synthtool
```
git checkout automatically-detect-when-samples-require-custom-dockerfile
docker build -f docker/owlbot/python/Dockerfile -t customdockerfile . 
```

Once the docker image was built, I ran the following commands in the root of python-recaptcha-enterprise.
```
docker run --user $(id -u):$(id -g) --rm -v $(pwd):/repo -w /repo customdockerfile
```

### CODE REVIEW

**Comment by parthea** [comment]
@busunkim96, Please could you review/approve?

**Comment by busunkim96** [comment]
@parthea I think this will break google-auth as it has an extra `noxfile.py` without a `blacken` session. Is there a workaround for that? https://github.com/googleapis/google-auth-library-python/blob/main/system_tests/noxfile.py

**Comment by busunkim96** [comment]
Whoops, please disregard my now deleted comment, I mixed this up with another PR.

File: synthtool/gcp/common.py [hunk 1]
```diff
@@ -266,6 +266,11 @@ def py_library(self, **kwargs) -> Path:
...
             kwargs["is_google_cloud_api"] = True
 
+        # If Dockerfile exists in .kokoro/docker/samples, add kwargs to
+        # signal that a custom docker image should be used when testing samples.
+        if Path(".kokoro/docker/samples/Dockerfile").exists():
+            kwargs["custom_samples_dockerfile"] = True
+
         # Assume the python-docs-samples Dockerfile is used for samples by default
         if "custom_samples_dockerfile" not in kwargs:
             kwargs["custom_samples_dockerfile"] = False
```
Comment (busunkim96) [inline synthtool/gcp/common.py:276]: Can the logic above be consolidated? Maybe `kwargs["custom_samples_dockerfile"] = Path("....").exists()`?

Comment (parthea) [reply synthtool/gcp/common.py]: Done.

**Review by busunkim96 — APPROVED** [review]

**Review by chingor13 — APPROVED** [review]

**Pull request merged**